### PR TITLE
Add user index page to front end.

### PR DESCRIPTION
**Pull request by DomPSU**
I also added a lot of comments so others can follow along.

### CODE REVIEW

File: src/components/users/UsersIndex.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import React, { Component, useReducer } from 'react';
+import Table from '../Table';
+
+// library for AJAX functions
+const axios = require('axios');
```
Comment (dlimx) [inline src/components/users/UsersIndex.js:5]: `import axios from 'axios';`

File: src/components/users/UsersIndex.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  
+  // after component mounts ( mount is React term for after the initital render)
+  componentDidMount() {
+
+    // call function to send AJAX to backend
+    this.getUsers();
+  }
+
+  getUsers() {
```
Comment (dlimx) [inline src/components/users/UsersIndex.js:33]: If you use an arrow function:

`getUsers = () => {}`

You don't need to call `.bind` as arrow functions preserve context

File: src/components/users/UsersIndex.js (hunk 1 above)
Comment (dlimx) [reply src/components/users/UsersIndex.js:5]: I'll abstract our API into a library - that way we can add authentication tokens

File: src/components/users/UsersIndex.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  // This render function is called when component first renders
+  // e.g. user clicks on link or when setState updates state
+  render() {
+
+    // set index from state
+    const { index } = this.state
+
+    // set values as empty arrays for initial render or AJAX fail
+    let tableHeaders = [];
+    let tableRows = [];
```
Comment (dlimx) [inline src/components/users/UsersIndex.js:55]: Why not `const`?

File: src/components/users/UsersIndex.js (hunk 2 above)
Comment (DomPSU) [reply src/components/users/UsersIndex.js:33]: When I change 29 to "getUsers = () => {}" , I get the error:  'getUsers' is not definded. Or should I do something else?

File: src/components/users/UsersIndex.js (hunk 3 above)
Comment (DomPSU) [reply src/components/users/UsersIndex.js:55]: in lines 59-60 I am reassigning them if the AJAX call was succesful. Is there a better way?

File: src/components/users/UsersIndex.js (hunk 1 above)
Comment (DomPSU) [reply src/components/users/UsersIndex.js:5]: Implemented change. Okay cool that will help alot.

File: src/components/users/UsersIndex.js (hunk 3 above)
Comment (dlimx) [reply src/components/users/UsersIndex.js:55]: I don't think this even needs reassigning, you could do something like: 
```
const tableHeaders = index.length ? Object.keys(index[0]) : []
```

And why reassign `index` to `tableRows` when both are arrays and you'll be using both the same way?

Comment (DomPSU) [reply src/components/users/UsersIndex.js:55]: Thanks I always forget about ternary operators haha.

Comment (DomPSU) [reply src/components/users/UsersIndex.js:55]: Yea I changed tableRows to index too. I think I did that cause I was getting use to eslint.

File: src/components/users/UsersIndex.js (hunk 2 above)
Comment (dlimx) [reply src/components/users/UsersIndex.js:33]: Ah, we're missing `babel-eslint` as our parser - feel free to leave as is and we can change it in a follow up

Comment (dlimx) [reply src/components/users/UsersIndex.js:33]: `babel` is what JavaScript uses to transform things like React syntax and modern JS to older ones, so we need to add the parser to `aslant`

**Pull request merged**